File: Python_codes/InsertionAtPosition/insert_at_sorted_list.py
print("START CODE")

L = [10,20,30,40,50,5]

print(" Before  L : ",L)

# assumption :  all elements except last in a list are sorted
# Key == last element in array
key = L[len(L) - 1]
print("key :  ",key)

i = len(L) - 2
print("index i :  ",i)

# 
while i > -1 and L[i] > key :
    L[i+1] = L[i]
    i = i - 1

L[i+1] = key

print(" After  L : ",L)

File: Java_codes/Sorting/src/insertionsort/InsertAtSortedList.java

public class InsertAtSortedList {
	public static void main(String[] args) {
		System.out.println("Start demo");

		int[] L = { 10, 20, 30, 40, 50, 25 };

		System.out.print("BEFORE List :");
		for(int element : L) {
			System.out.print(element +" ");
		}
		
		Integer key = L[L.length - 1];
		System.out.println("\n----");
		System.out.println("Key : "+ key);
		int i = L.length - 2;
		System.out.println("i : "+ i);
		System.out.println("----");

		
		while(i > -1 && L[i] > key)
		{
			L[i+1] = L[i];
			i = i-1;
		}
		L[i+1] = key;
		System.out.print("After List : ");
		for(int element : L) {
			System.out.print(element + " ");
		}

	}
}
